moscar.go

```diff
@@ -1,124 +0,0 @@
-package main
-
-import (
-	"fmt"
-	"github.com/micromdm/squirrel/munki/datastore"
-	"github.com/micromdm/squirrel/munki/models"
-	"html/template"
-	"io"
-	"log"
-	"net/http"
-	"time"
-)
-
-const STATIC_URL string = "/static/"
-const STATIC_ROOT string = "static/"
-const REPO_PATH string = "/Users/Shared/repo"
-
-type Context struct {
-	Static string
-	Prods  []ParsedPkginfo
-}
-
-// Template expects icon, 'product' name, description, and version, which becomes the download URL
-type ParsedPkginfo struct {
-	Icon        string
-	Name        string
-	Descript    string
-	Version     string
-	DownloadURL string
-}
-
-func Parse(in_pkginfo *models.PkgsInfo, ppi *ParsedPkginfo) *ParsedPkginfo {
-	ppi.Icon = in_pkginfo.IconName
-	ppi.Name = in_pkginfo.Name
-	ppi.Descript = in_pkginfo.Description
-	ppi.Version = in_pkginfo.Version
-	ppi.DownloadURL = in_pkginfo.PackageCompleteURL
-	return ppi
-}
-
-func Home(w http.ResponseWriter, req *http.Request) {
-	repo := datastore.SimpleRepo{Path: REPO_PATH}
-	log.Print("found repo: ", repo)
-	catalog, err := repo.AllPkgsinfos()
-	if err != nil {
-		log.Fatal("faux-catalog building error: ", err)
-	}
-	allcat := *catalog
-	prods := make([]ParsedPkginfo, 0)
-	for _, pkginfo := range allcat {
-		pi := &ParsedPkginfo{}
-		ppi := Parse(pkginfo, pi)
-		prods = append(prods, *ppi)
-	}
-	context := Context{Prods: prods}
-	render(w, "gomoscar.html", context)
-}
-
-func render(w http.ResponseWriter, tmpl string, context Context) {
-	context.Static = STATIC_URL
-	tmpl_list := []string{"templates/gomoscar.html",
-		fmt.Sprintf("templates/%s", tmpl)}
-	t, err := template.ParseFiles(tmpl_list...)
-	if err != nil {
-		log.Print("template parsing error: ", err)
-	}
-	err = t.Execute(w, context)
-	if err != nil {
-		log.Print("template executing error: ", err)
-	}
-}
-
-func StaticHandler(w http.ResponseWriter, req *http.Request) {
-	static_file := req.URL.Path[len(STATIC_URL):]
-	if len(static_file) != 0 {
```

Comment:
Failing early makes more sense. Instead of writing `if err == nil { // do the right thing } `, write `if err != nil { return err } ` and then do the correct thing below. 
Same thing with the `static_file`. Check if it's an empty string first and then return `NotFound` and then do the case where everything is fine below.